app/static/src/app/serialise.ts

```diff
@@ -51,6 +52,21 @@ function serialiseDiscreteResult(result: OdinRunResultDiscrete | null): Serialis
     } : null;
 }
 
+const serialiseAdvancedSettings = (settings: AdvancedSettings) => {
+    return Object.fromEntries(
+        Object.entries(settings).map(([key, value]) => {
+            if (!value.standardForm) {
+                if (!Number.isFinite(value.defaults)) {
+                    const copy = { ...value } as any;
+                    copy.defaults = "Infinity";
+                    return [key, copy];
+                }
+            }
+            return [key, value];
+        })
+    ) as SerialisedAdvancedSettings;
+};
```

Comment:
See above comment, I didn't think the values actually needed to be Infinity  ... and anyway, I thought that was the displayed placeholder? I think we can just stick to null for the actual default values can't we? or @richfitz is that a change to the runner required to accept nulls?

Reply:
i did try but it was returning errors when i input null? maybe there is support for defaulting if undefined but not if null in odinjs? ill have to take a look at the code or try this out!

Reply:
okay so undefined default value works but null does not! are we just planning to leave the placeholder empty for the maxStepSize then? (this was the only one that has a default of infinity)

Reply:
sorry just saw your comment above about current end time, so these values I thought would always be hard coded in, we can pull from store like the current endtime but i wonder if that is more complex than we need it to be, the components get placeholders from their props and get it from the object, we dont have much of a distinction between the advanced settings but we would have to have a special case for just the max step size and nothing else, i wonder if leaving it blank is alright? and maybe using the help tooltips in some way?